crates/extensions/c8y_mapper_ext/src/converter.rs

```diff
@@ -318,33 +318,19 @@ impl CumulocityConverter {
 
     pub fn process_alarm_messages(
         &mut self,
-        topic: &Topic,
-        message: &Message,
+        source: &EntityTopicId,
+        input: &Message,
+        alarm_type: &str,
     ) -> Result<Vec<Message>, ConversionError> {
-        if topic.name.starts_with("tedge/alarms") {
-            let mut mqtt_messages: Vec<Message> = Vec::new();
-            self.size_threshold.validate(message)?;
-            let mut messages = self.alarm_converter.try_convert_alarm(message)?;
-            if !messages.is_empty() {
-                // When there is some messages to be sent on behalf of a child device,
-                // this child device must be declared first, if not done yet
-                let topic_split: Vec<&str> = topic.name.split('/').collect();
-                if topic_split.len() == 5 {
-                    let child_id = topic_split[4];
-                    add_external_device_registration_message(
-                        child_id.to_string(),
-                        &mut self.children,
-                        &mut mqtt_messages,
-                    );
-                }
-            }
-            mqtt_messages.append(&mut messages);
+        self.size_threshold.validate(input)?;
+
+        if let Some(entity) = self.entity_store.get(source) {
+            let mqtt_messages = self
+                .alarm_converter
+                .try_convert_alarm(entity, input, alarm_type)?;
             Ok(mqtt_messages)
-        } else if topic.name.starts_with(INTERNAL_ALARMS_TOPIC) {
-            self.alarm_converter.process_internal_alarm(message);
-            Ok(vec![])
         } else {
-            Err(ConversionError::UnsupportedTopic(topic.name.clone()))
+            Ok(vec![])
```

Comment:
I'd return an error about unknown entity so that this error is properly reported on `tedge/errors`.

Reply:
As @didier-wenzek, changing from `EntityMetadata` to `EntityTopicId` solved this. https://github.com/thin-edge/thin-edge.io/pull/2267/commits/6536b54e6401841d88d0b20c3e24650ff8d9e894